e2e/tests/attendance-tests.spec.ts

```diff
@@ -1,360 +1,278 @@
+    page.getByRole("heading", { name: "School Class 2B" }),
+  ).toBeVisible();
 
-    /*
-      Verify list of classes is displayed
-    */
-    await page.waitForSelector("mat-card-header");
-    const classList = page.locator("mat-card-header");
-    const count = await classList.count();
-    expect(count).toBeGreaterThan(0);
+  await expect(page.getByText("1 / 6")).toBeVisible();
+  await page.getByRole("button", { name: "Present" }).click();
 
-    /*
-      Mark attendance for participants dynamically
-    */
-    const firstClass = page.locator("mat-card-header").first();
-    await firstClass.evaluate((el: HTMLElement) => el?.click()); // Force the DOM click
-
-    let totalParticipants = 0;
-
-    const pageIndicator = page.locator("text=/\\d+\\s\\/\\s\\d+/"); // "1 / 5" pattern
-    // Extract total participants from "x / y" format
-    if (await pageIndicator.isVisible()) {
-      const text = await pageIndicator.textContent();
-      const match = text?.match(/(\d+)\s\/\s(\d+)/); // Extract current/total numbers
-      if (match) {
-        totalParticipants = parseInt(match[2]); // Total participants
-      }
-    }
-    const ReviewDetailsBtn = page.getByRole("button", {
-      name: "Review Details",
-    });
-
-    // Loop to mark attendance dynamically until the "Review Details" button is visible
-    while (!(await ReviewDetailsBtn.isVisible())) {
-      const randomIndex = Math.floor(Math.random() * 4); // Random index for attendance buttons
-      const attendanceButton = page.getByRole("paragraph").nth(randomIndex);
-      await attendanceButton.click();
-      await page.waitForTimeout(1000); // Wait for the next participant to load
-    }
+  await expect(page.getByText("2 / 6")).toBeVisible();
+  // FIXME: There are two buttons with name Absent. This is caused by the
+  // roll-call-tab component which briefly shows both tabs when animating the
+  // transition. Can we globally disable animations? Or can we find a way to
+  // disable the outgoing tab
+  await expect(page.getByRole("button", { name: "Absent" })).toHaveCount(1);
+  await page.getByRole("button", { name: "Absent" }).click();
 
-    await expect(ReviewDetailsBtn).toBeVisible();
-    await ReviewDetailsBtn.click();
+  await expect(page.getByText("3 / 6")).toBeVisible();
+  await expect(page.getByRole("button", { name: "Late" })).toHaveCount(1);
+  await page.getByRole("button", { name: "Late" }).click();
 
-    /*
-      Review popup: Verify the status are editable and can be updated
-    */
-    const EditStatusBtn = page
-      .locator(
-        "app-attendance-status-select > .mat-mdc-form-field > .mat-mdc-text-field-wrapper > .mat-mdc-form-field-flex",
-      )
-      .first();
-    await EditStatusBtn.click();
-
-    const statusOptions = page.getByText("Present Absent Late Excused");
-    const randomIndex = Math.floor(
-      Math.random() * (await statusOptions.count()),
-    );
-    const selectedStatus = statusOptions.nth(randomIndex);
```

Comment:
Using Math.random makes the tests non-deterministic which makes them hard to follow and debug because each test run is different. Instead, just [chose an option](https://xkcd.com/221/) when you run it.